### PR TITLE
Throws RuntimeException with more accurate error message if buildActivity passed class is abstract

### PR DESCRIPTION
**Pull request by utzcoz**
Close https://github.com/robolectric/robolectric/issues/8997.

### CODE REVIEW

**Review by MGaetan89 — APPROVED** [review]

File: integration_tests/ctesque/src/sharedTest/java/android/app/ActivityTest.java [hunk 1]
```diff
@@ -77,4 +78,16 @@ public void whenNotSetOnActivityInManifest_activityGetsThemeFromApplicationInMan
...
+  public void launchActivity_abstractActivity_throwsRuntimeException() {
+    boolean catchException = false;
+    try (ActivityScenario<AbstractTestActivity> scenario =
+        ActivityScenario.launch(AbstractTestActivity.class)) {
+      assertThat(scenario).isNull();
+    } catch (RuntimeException exception) {
+      catchException = true;
+    }
+    assertThat(catchException).isTrue();
+  }
```
Comment (MGaetan89) [inline integration_tests/ctesque/src/sharedTest/java/android/app/ActivityTest.java:92]: Wouldn't it be enough to do the followng:
```suggestion
  @Test(expected = RuntimeException.class)
  public void launchActivity_abstractActivity_throwsRuntimeException() {
    try (ActivityScenario<AbstractTestActivity> scenario =
                 ActivityScenario.launch(AbstractTestActivity.class)) {
      assertThat(scenario).isNull();
    }
  }
```

File: robolectric/src/test/java/org/robolectric/shadows/ShadowActivityTest.java [hunk 2]
```diff
@@ -1501,6 +1501,22 @@ public void buildActivity_optionBundleWithInvalidNonDefaultDisplaySet_launchesOn
...
+    Throwable throwable =
+        assertThrows(
+            RuntimeException.class,
+            () -> {
+              ActivityController<AbstractTestActivity> controller =
+                  Robolectric.buildActivity(AbstractTestActivity.class, null);
+              // This line will not be executed.
+              assertThat(controller).isNull();
+            });
+    assertThat(throwable).isInstanceOf(RuntimeException.class);
```
Comment (MGaetan89) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowActivityTest.java:1515]: `assertThrows` returns a throwable of the type you provide (here `RuntimeException`).
So line 1506 could be `RuntimeException throwable =`, and this check can be removed.

File: testapp/src/main/java/org/robolectric/testapp/AbstractTestActivity.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package org.robolectric.testapp;
+
+import android.app.Activity;
+
+/** Abstract test [Activity] for test purpose. */
```
Comment (MGaetan89) [inline testapp/src/main/java/org/robolectric/testapp/AbstractTestActivity.java:5]:
```suggestion
/** Abstract test {@link Activity} for test purpose. */
```

**Review by david-allison — APPROVED** [review]
LGTM

Only question is the double-definition of `AbstractTestActivity`:
* `org.robolectric.testapp.AbstractTestActivity` 
* `org.robolectric.shadows.ShdowActivityTest.AbstractTestActivity`

I presume this is a requirement of your testing frameworks

Thanks!

**Pull request merged**